pkg/tagger/local/tagstore.go

```diff
@@ -416,7 +416,7 @@ func (e *entityTags) computeCache() {
 
 func (e *entityTags) isEmpty() bool {
 	for _, tags := range e.sourceTags {
-		if !tags.isEmpty() {
+		if !tags.expiryDate.IsZero() || !tags.isEmpty() {
```

Comment:
I think we should rename the method, something like `func (e *entityTags) shouldRemove() bool`